lib/spack/docs/advanced_topics.rst

```diff
@@ -0,0 +1,263 @@
+.. Copyright Spack Project Developers. See COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+.. _toolchains:
+
+=============================
+Defining and Using Toolchains
+=============================
+
+Spack can be configured to associate certain combinations of specs for
+easy reference on the command line and in config and environment
+files. These combinations are called ``toolchains``, because their
```

Comment:
Can we start with a motivating example instead of with the definition? e.g.:

> Spack lets you specify compilers on the CLI with, e.g., `%gcc` or `%c,cxx=clang %fortran=gcc`, and you can specify flags with `cflags`, `cxxflags`, and `fflags`. Depending on how complex your compiler setup is, it can be cumbersome to specify all of your preferences on the CLI. Spack has a special type configuration called  *toolchains*, which let you encapsulate the configuration for compilers, other libraries, and flags into a single name that you can reference as though it were one option.